test/runtest.py

```diff
@@ -1538,7 +1538,8 @@ def testGlobalStateAPI(self):
     task_table = ray.global_state.task_table()
```

Comment:
Let's add another test in `runtest.py` that tests calling `ray.init` with `num_redis_shards=10` or `20` or some large number and then runs some remote functions and gets the results. I don't think we're testing the changes to the `ray.init` code path anywhere.